### ISSUE TITLE
Clickjacking and disabled X_FRAME_OPTIONS for authorize endpoint

### ISSUE THREAD

**Issue by ggjersund** [juanifioren/django-oidc-provider#267]
Are X_FRAME_OPTIONS=False for the authorize endpoints? I'm wondering because I'm performing a silent refresh for an API, but I don't want my sign-in and sign-up forms to have iframe allowed.

**Comment by juanifioren** [comment]
Hi today we are relying on [Django Clickjacking Protection](https://docs.djangoproject.com/en/dev/ref/clickjacking/). By default, `X_FRAME_OPTIONS = 'SAMEORIGIN'`.

**Comment by ggjersund** [comment]
@juanifioren My bad. I think I missed a technicality when it comes to silent refresh on front-end clients. They are of course not loading the authorize endpoints inside an iframe at the client, but are simply using a default get request from the client, which redirects back to an iframe at the client.

**Comment by juanifioren** [comment]
I do a little research about this: _authorization endpoints should be a different URL than the consent endpoint_. Some providers use different urls others the same.

So maybe you can create a feature request about splitting consent from auth request.

**Comment by ggjersund** [comment]
@juanifioren Feature request added at #268

**Comment by q3aiml** [comment]
Maybe I am misunderstanding this and #268, but from my experience silent renew (authorize iframe with `prompt=none`) already works. X-Frame-Options does not block a redirect -- it only blocks displaying content. When I do see framing fail due to X-Frame-Options it is often a client configuration issue such as an unregistered redirect_uri.

**Comment by ggjersund** [comment]
@q3aiml So prompt=None basically ignores X-Frame-Options, by disabling the UI?

**Comment by q3aiml** [comment]
Yeah, more or less. X-Frame-Options has no impact because the browser follows redirects regardless. And as you say prompt=none ensures the response is always a redirect by disabling the UI

**Comment by ggjersund** [comment]
@q3aiml, then I have completely misunderstood the use of X-Frame-Options, I thought it disabled all possibilities of framing the website, and prevented any actions to be performed (even redirect). However, if this is the case, I guess one can close these two issues.
